libs/domain/feature-loaders/src/snapshot/SnapshotProposalLoader.ts

```diff
@@ -0,0 +1,187 @@
+import { Data, Nested, NonEmptyProperty, OptionalNumberArrayOf, OptionalObjectRef, OptionalProperty, OptionalStringArrayOf, Property, RequiredArrayRef, RequiredStringArrayOf } from "@banklessdao/util-schema";
+import { DocumentNode } from "graphql";
+import gql from "graphql-tag";
+import * as t from "io-ts";
+import * as TE from "fp-ts/TaskEither";
+import { GraphQLDataLoaderBase } from "../base/GraphQLDataLoaderBase";
+import { BATCH_SIZE } from "../defaults";
+import { withMessage } from "io-ts-types";
+
+export const URL = "https://hub.snapshot.org/graphql";
+
+export const makeQUERY: (space:String)=>DocumentNode = (space) => {
+    space = `"${space}"` // So it gets parsed correctly
+    return gql`
+        query snapshotProposals($limit: Int,$cursor: Int) {
+            proposals (
+                first: $limit,
+                where: {
+                    space_in: [${space}],
+                },
+                skip:$cursor,
+                orderBy: "created",
+                orderDirection: asc
+            ) {
+                id
+                strategies {
+                    name
+                    params
+                }
+                title
+                body
+                choices
+                created
+                start
+                end
+                snapshot
+                state
+                author
+                space {
+                    id
+                    name
+                }
+            }
+    }
+    `;
+}
+
+// Helper for optional fields
+const optional = <T extends t.Mixed>(x:T)=>t.union([x,t.undefined])
+
+export const ProposalCodec = t.strict({
+    id: t.string,
+    author: t.string,
+    created: t.number,
+    space: optional(t.strict({
+        id: t.string,
+        name: t.string
+    })),
+    type: optional(t.string),
+    strategies: withMessage(t.array(t.strict({
+        name:t.string,
+    })),()=>"strategies failed"),
+    title: t.string,
+    body: optional(t.string),
+    choices: t.array(t.string),
+    start: t.number,
+    end: t.number,
+    snapshot: t.string,
+    state: t.string,
+    link: optional(t.string),
+    scores: optional(t.array(t.number)),
+    votes: optional(t.number)
+});
+
+export const ProposalsCodec = t.strict({
+    proposals: t.array(ProposalCodec),
+});
+
+export type Proposals = t.TypeOf<typeof ProposalsCodec>;
+
+const INFO = {
+    namespace: "snapshot",
+    name: "Proposal",
+    version: "V1",
+};
+@Nested()
+class Space {
+    @OptionalProperty()
+    name: string
+    @NonEmptyProperty()
+    id: string
+}
+
+@Nested()
+class Strategy {
+    @NonEmptyProperty()
+    name:string
+}
+
+@Data({
+    info: INFO,
+})
+export class Proposal {
+    @NonEmptyProperty()
+    id: string
+    @NonEmptyProperty()
+    author: string
+    @NonEmptyProperty()
+    created: number
+    @OptionalObjectRef(Space)
+    space: Space | undefined
+    @OptionalProperty()
+    type:string | undefined
```

Comment:
Is there a reason for using `type: string | undefined;` instead of `type?: string;`?

Reply:
Nope, I changed it to use the "?" notation instead